### PR TITLE
[TS] LPS-161392

### PR DESCRIPTION
**Pull request by jcampoy**
https://issues.liferay.com/browse/LPS-161392

### CODE REVIEW

**Comment by jcampoy** [comment]
ci:test:relevant

**Comment by jcampoy** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -2046,7 +2046,7 @@ private void _insertIntoTable(
 		for (ObjectField objectField : objectFields) {
 			Object value = values.get(objectField.getName());
```
Comment (seltong) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2047]: Hi @jcampoy, can we move [the line](https://github.com/liferay-objects/liferay-portal/pull/964/commits/9b8b08f74a24b599820ce715afc40d2249491ba5#diff-520634a879bf903adc87e7625aa3ffd7470975dac938d58c51c40d478b352dabR2047) to before line 2068 like the code below?


```
	if (Objects.equals(
			objectField.getRelationshipType(),
			ObjectRelationshipConstants.TYPE_ONE_TO_ONE)) {

		Object value = values.get(objectField.getName());

		_validateOneToOneInsert(
			objectField.getDBColumnName(), GetterUtil.getLong(value),
			dynamicObjectDefinitionTable);
	}
```

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -2111,7 +2111,7 @@ private void _insertIntoTable(
 			for (ObjectField objectField : objectFields) {
 				Object value = values.get(objectField.getName());
```
Comment (seltong) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2099]: Hi @jcampoy, can we move [the line](https://github.com/liferay-objects/liferay-portal/pull/964/commits/9b8b08f74a24b599820ce715afc40d2249491ba5#diff-520634a879bf903adc87e7625aa3ffd7470975dac938d58c51c40d478b352dabR2112) to before line 2121 like the code below?

```
	for (ObjectField objectField : objectFields) {
		if (!values.containsKey(objectField.getName())) {
			continue;
		}

		Column<?, ?> column = dynamicObjectDefinitionTable.getColumn(
			objectField.getDBColumnName());

		Object value = values.get(objectField.getName());

		_setColumn(
			preparedStatement, index++, column.getSQLType(), value);
	}
```

**Review by seltong — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 1 above)
Comment (jcampoy) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java]: Hi @seltong , see my changes [a19f7b4](https://github.com/liferay-objects/liferay-portal/pull/964/commits/a19f7b46aea99990bdf36026e24a595adf50407c), if you think it is better to continue saving the value in a new Object, please, go ahead and change it and resend, we would like to fix it asap.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 2 above)
Comment (jcampoy) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java]: Hi @seltong , see my changes [a19f7b4](https://github.com/liferay-objects/liferay-portal/pull/964/commits/a19f7b46aea99990bdf36026e24a595adf50407c), if you think it is better to continue saving the value in a new Object, please, go ahead and change it and resend, we would like to fix it asap.

**Review by seltong — APPROVED** [review]

**Comment by jcampoy** [comment]
ci:test:relevant

**Comment by jcampoy** [comment]
Hi @seltong , it seems the failed tests are not related, how can we progress with this issue?
BR

**Comment by seltong** [comment]
Hi @jcampoy. Could you please manually submit a pull request to Brian's repository with a message explaining that it was passed my review and the ci relevant failures are unrelated?

**Comment by seltong** [comment]
Hi @jcampoy I saw that you removed [this condition](https://github.com/liferay-objects/liferay-portal/pull/964/files#diff-520634a879bf903adc87e7625aa3ffd7470975dac938d58c51c40d478b352dabL2050) and I don't know if we can do it that way. I need to review your pull request again. Sorry about that, I will review and update you asap.

Thank you

**Comment by jcampoy** [comment]
Hey @seltong , ok, if this is a problem we can introduce the condition again, because it does not affect for solving this issue and I only was cleaning the code.

I removed it because this method *_insertIntoTable* is always used from methods that already have been validated the required values through the method _validateValues , take a look to the code and if you have any doubt, let me know..

**Comment by seltong** [comment]
Very nice, @jcampoy and thank you to clean the code.
Your fix looks good to me, but [this test](https://testray.liferay.com/home/-/testray/case_results/1963541435?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D1963539855%26cur%3D1%26delta%3D200%26tab%3Druns%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc) is failing on the ci relevant. Could you please take a look at this?

**Comment by jcampoy** [comment]
@seltong the. problem was to assume that _validateValues method was validating every field in the definition, this is not true and it is only validating the provided values.

I tested without that commit and one of the failing tests works again, I rebased this pull without that change and send it again, could you take a look? https://github.com/liferay-objects/liferay-portal/pull/1022

**Comment by seltong** [comment]
Hi, @jcampoy. Thanks for your hard work on this.
I will take a look at this as soon as possible.

**Pull request closed without merging**